### PR TITLE
document and test ways to access Variables in a Model

### PR DESCRIPTION
**Pull request by whoburg**
@bqpd this documents and tests variable access. Ready for review.

### CODE REVIEW

**Comment by bqpd** [comment]
test models please

File: docs/source/examples/model_var_access.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        f = Variable("f", 20, "lb/hp", "mass per unit power")
+        Pmax = Variable("P_{max}", "hp", "max output power")
+        return [m >= f*Pmax]
+
+
+class PowerSystem(Model):
+    "A battery powering a motor"
+    def setup(self):
+        # self.battery = Battery()
+        # self.motor = Motor()
```
Comment (bqpd) [inline docs/source/examples/model_var_access.py]: should these lines be deleted?

File: docs/source/examples/model_var_access.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        components = [Battery(), Motor()]
+        m = Variable("m", "lb", "mass")
+        return [components,
+                m >= sum(comp.topvar("m") for comp in components)]
+
+PS = PowerSystem()
+print "Getting the only var 'E': ", PS["E"]
+print "The top-level var 'm': ", PS.topvar("m")
```
Comment (bqpd) [inline docs/source/examples/model_var_access.py]: After writing this, how do you feel about the `topvar` syntax?

File: docs/source/modelbuilding.rst [hunk 3]
```diff
@@ -13,6 +13,19 @@ Variables created in a ``setup`` method are added to the model even if they are
 
 At the end of this page a detailed example shows this technique in practice.
 
+Accessing Variables in Models
```
Comment (bqpd) [inline docs/source/modelbuilding.rst]: Nice. I really like this method of documenting things; I think we should convert much of the rest of the documentation to this style.

File: docs/source/examples/model_var_access.py (hunk 1 above)
Comment (whoburg) [inline docs/source/examples/model_var_access.py]: yes, good call

File: docs/source/examples/model_var_access.py (hunk 2 above)
Comment (whoburg) [inline docs/source/examples/model_var_access.py]: seems good to me

File: docs/source/modelbuilding.rst (hunk 3 above)
Comment (whoburg) [inline docs/source/modelbuilding.rst]: just so I understand, what do you mean by 'this style'?

**Comment by whoburg** [comment]
test models please

**Comment by whoburg** [comment]
test models please

**Comment by whoburg** [comment]
ok, responded to review. The only model failing is turbofan, which wasn't caused by this. Merging.

File: docs/source/modelbuilding.rst (hunk 3 above)
Comment (bqpd) [inline docs/source/modelbuilding.rst]: Documenting everything through in-practice (useful) examples.

**Pull request merged**